[2] Support for strings of multiple intensifiers

Provinite opened the pull request:
### Description
Add support for strings of repeated intensifiers with or without terminator by consuming more of the message if firstWord and secondWord are the same intensifier.

"I am very very tired" -> "Hi Mr(s) Tired, can I call you Very?"
"I am very very very" -> "Hi Very, I'm Fido!"

### JIRA Task Links
https://akitabox.atlassian.net/browse/WEBAPP-5178

### Related PR Links
N/A

### Post Deployment Checklist
- Add script to fido

### Dev Owner Checklist
_Finish these before moving the JIRA task to `Ready for Review`_
- [x] List manual testing workflows below
- [x] Perform reviewer checklist on your changes 

### Manual Testing workflows
- N/A

### Reviewer Checklist
_Finish these before merging this PR and moving the JIRA task to `Merged`_
- [ ] Read the changes looking for syntax errors, copypasta, and appropriate commenting
- [ ] Read the changes reviewing algorithms
    - [ ] New functionality is added with sound algorithms
    - [ ] Bugs are fixed using the appropriate solution given the time constraints and business priorities
- [ ] Ensure that automated test cases adequately cover the changes 
    - [ ] Tests to cover newly added functionality 
    - [ ] Tests to prevent the bug from being shipped and reproduced in the future
    - [ ] Are there any missing test cases that should be added?
- [ ] Manually test functionality not covered by automated tests (should be listed above in testing workflows)
    - [ ] Are there workflows that should be tested manually that are not listed?
- [ ] Does the finished product meet acceptance criteria listed in the applicable JIRA issue?
- [ ] **Are the changes production-ready and bug free?**

Commit:
feat(dadbot): add smarter responses for messages with multiple intensifiers

add appropriate tests and functionality for messages like "I am very very hungry"

fixes #2

bowernite commented on test/dadbot-test.coffee:
```diff
@@ -31,12 +31,26 @@ describe 'DadBot', ->
   it 'responds to "I AM HuNgRy"', ->
     @room.user.say('bob', 'I AM HuNgRy').then =>
       expect(@room.messages).to.eql [
-        ['bob', 'I AM HuNgRy'],
```
Is this a coffeescript thing (not needing a comma between arrays if they're separated by a line)?

Provinite replied:
yeah, I just tweaked that so the styling would fit better with the other tests

bowernite commented on src/dadbot.js:
```diff
@@ -76,12 +76,42 @@ module.exports = (robot) => {
+            wordStack.pop();
+            // consume to end of string looking for a terminator
+            while (firstWord == secondWord && wordStack.length > 0) {
+                secondWord = wordStack.pop();
```
`secondWord` looks to no longer just represent the "second word", but rather a "terminator. Maybe warrants a variable rename, or a new variable altogether (that is initialized to the value of `secondWord`, but can change values, like it does in this line).

bowernite commented on src/dadbot.js:
```diff
@@ -76,12 +76,42 @@ module.exports = (robot) => {
+        let secondWord = lowerCase(res.match[2]);
+
+        // caught something like "i am very very tired"
+        if (isIntensifier(firstWord) && firstWord == secondWord) {
```
What if, instead of just looking for duplicate first and second words, you keep look at if the second word is an intensifier as well?

For example, it would then handle the case where two intensifiers (that are different) come in a row:

`"I am so extremely tired"` --> `"Hi Mr(s) Tired, can I call you So Extremely?"

I think right now as implemented, this would still output:
`"I am so extremely tired"` --> `"Hi Mr(s) Extremely, can I call you So?"`

bowernite commented on src/dadbot.js:
```diff
@@ -76,12 +76,42 @@ module.exports = (robot) => {
-        let firstWord = capitalize(res.match[1]);
-        let secondWord = capitalize(res.match[2]);
+        let lowerCase = (str) => {
+            if (str == undefined) return undefined;
```
This check could be a bit more robust and catch `null` cases.

```javascript
if (!str) return str;
```

Commit:
feat(dadbot): improve smartness for multiple intensifiers

add support for strings of multiple different intensifiers, improve
some variable naming, offload some work to helper functions

fixes #2

bowernite commented:
Code looks good!

Looks like you're still missing some things when it comes to Travis. From the Jira ticket:

```
4. *Configure your repo to use Travis CI*
Add a travis.yml file to your hubot function repo. Configure Travis to execute `npm test` to lint and run any automated tests.
```

So -- you need to:

1. Get a Travis CI account, and hook it up to this repository. Every commit you push should trigger a build on Travis, and display the result next to it (green checkmark or red X). https://docs.travis-ci.com/
2. Modify your `.travis.yml` to lint and test on every build.

Provinite commented:
See https://github.com/Provinite/hubot-dadbot/pull/7

Provinite commented:
Rebased from dev with travis integration.

bowernite commented:
Looks good to me!
